### PR TITLE
Add one offs to review email list page

### PR DESCRIPTION
**Pull request by whabanks**
# Summary | Résumé
This PR adds functionality to include one off emails in the "Review email addresses" page. 

- Link to the notification that failed 
- Instead of the job name we use the template name.
- Only display the first 20 one off sends & link to the reports page with the problem email filters enabled

image

# Test instructions | Instructions pour tester la modification
1. Send a one off email by the API or the UI that will lead to a bounce
2. Navigate to the `review email list` page
3. Note that one off's are now included in the list of addresses to review
4. Clicking on an email address in the list should take you to the notification page
5. Clicking "See all problem addresses" should link to the reports page with the "Only show problem addresses" filter enabled.

### CODE REVIEW

**Review by jzbahrai — APPROVED** [review]
Followed the QA steps and tested this. Looks good to me.

File: app/templates/views/dashboard/review-email-list.html [hunk 1]
```diff
@@ -47,9 +48,21 @@ <h2 class="heading-medium mt-8">{{ _('Problem emails over the past 24 hours') }}
...
+                    <a href="{{ url_for('.view_notification', service_id=current_service.id, notification_id=notification.id) }}">{{notification["to"]}}</a>
+                    <span>
+                        {{ _( ' in ')}} '{{ notification["template"]["name"] }}'
+                    </span>
+                </li>
+            {% endfor %}
             </ul>
+            {{ show_more(
+                url_for('.view_notifications', service_id=current_service.id, message_type='email', status='permanent-failure', pe_filter='true'),
+                _("See all problem email addresses")
```
Comment (amazingphilippe) [inline app/templates/views/dashboard/review-email-list.html:62]: We'll need to add this key in our translation file

**Review by amazingphilippe — COMMENTED** [review]
Found a couple missing translations. One of them seems to come from this PR, but the other one might have slipped through in earlier PRs:

Multiple problem addresses: 
Capture d’écran, le 2023-05-16 à 15 49 10

See all problem addresses:
Capture d’écran, le 2023-05-16 à 15 49 18


Otherwise, functionality looks good. Just noting that the design of the problem-emails page might evolve in 1.5. I still think we can merge for now.

File: app/templates/views/dashboard/review-email-list.html (hunk 1 above)
Comment (whabanks) [reply app/templates/views/dashboard/review-email-list.html:68]: Added!

**Review by andrewleith — COMMENTED** [review]
Looks great, one little note about clarity of bulk sends vs one-off sends when they are both on the review page below.

File: app/templates/views/dashboard/review-email-list.html [hunk 2]
```diff
@@ -46,16 +47,35 @@ <h2 class="heading-medium mt-8">{{ _('Problem emails over the past 24 hours') }}
...
                             {{ _( ' in ')}} '{{ job.original_file_name }}'
                         </span>
                     </li>
                 {% endif %}
             {% endfor %}
+            {% for notification in one_offs %}
+                <li>
+                    <a href="{{ url_for('.view_notification', service_id=current_service.id, notification_id=notification.id) }}">{{notification["to"]}}</a>
+                    <span>
+                        {{ _( ' in ')}} '{{ notification["template"]["name"] }}'
```
[the remark this answers is not in the document]
Comment (whabanks) [reply app/templates/views/dashboard/review-email-list.html]: I like this idea, it definitely makes the distinction between the two sending types more clear!

**Comment by jzbahrai** [comment]
Tested: 
Screenshot 2023-05-19 at 2 57 49 PM

**Review by jzbahrai — APPROVED** [review]

File: app/templates/views/dashboard/review-email-list.html [hunk 3]
```diff
@@ -46,16 +47,35 @@ <h2 class="heading-medium mt-8">{{ _('Problem emails over the past 24 hours') }}
...
                             {{ _( ' in ')}} '{{ job.original_file_name }}'
                         </span>
                     </li>
                 {% endif %}
             {% endfor %}
+            {% for notification in one_offs %}
+                <li>
+                    <a href="{{ url_for('.view_notification', service_id=current_service.id, notification_id=notification.id) }}">{{notification["to"]}}</a>
+                    <span>
+                        {{ _( ' using template ')}} '{{ notification["template"]["name"] }}'
```
Comment (amazingphilippe) [inline app/templates/views/dashboard/review-email-list.html:68]: We'll need a translation for this!

**Pull request merged**